[MIG]Se inicia la migración de l10n_ve_accountant.

OmarYepez29 opened the pull request:
Se cambia los currency_foreign_id por foreign_currency_id Link:https://binaural.odoo.com/web#id=55235&cids=2&menu_id=257&action=341&model=project.task&view_type=form

Ivan-Binaural committed:
FEAT #55256: l10n_ve_stock

Problema: Se necesita migrar este modulo, aparte uno de los cambios tiene influencia en otro modulo binaural_stock_accountant

Solución:se cambian y se actualizan de tree a list entre otros cambios a parametros

Tarea (Link):https://binaural.odoo.com/web#id=55256&cids=2&menu_id=975&action=341&model=project.task&view_type=form

Tarea de proyecto [x]
Ticket de soporte []

OmarYepez29 committed:
[MIG]Se inicia la migración de l10n_ve_accountant.

Se cambia los currency_foreign_id por foreign_currency_id
Link:https://binaural.odoo.com/web#id=55235&cids=2&menu_id=257&action=341&model=project.task&view_type=form

Ivan-Binaural committed:
FEAT #55256: l10n_ve_stock

Problema: Se necesita migrar este modulo, aparte uno de los cambios tiene influencia en otro modulo binaural_stock_accountant

Solución:se cambian y se actualizan de tree a list entre otros cambios a parametros

Tarea (Link):https://binaural.odoo.com/web#id=55256&cids=2&menu_id=975&action=341&model=project.task&view_type=form

Tarea de proyecto [x]

Ticket de soporte []

Ivan-Binaural committed:
MIG #55256: l10n_ve_stock

Problema: Se necesita migrar este modulo, aparte uno de los cambios tiene influencia en otro modulo binaural_stock_accountant

Solución:se cambian y se actualizan de tree a list entre otros cambios a parametros

Tarea (Link):https://binaural.odoo.com/web#id=55256&cids=2&menu_id=975&action=341&model=project.task&view_type=form

Tarea de proyecto [x]

Ticket de soporte []

OmarYepez29 committed:
[MIG/FIX]l10n_ve_accountant.

Link:https://binaural.odoo.com/web#id=55235&cids=2&menu_id=257&action=341&model=project.task&view_type=form

Ivan-Binaural committed:
Merge remote-tracking branch 'origin/18-mig-l10n_ve_accountant-#55235' into l10n_ve_stock_account_MIG#55257

Ivan-Binaural committed:
MIG #55257: l10n_ve_stock_account

Problema: Migrando el modulo a la version 18

Solución:

Tarea (Link):

Tarea de proyecto [x]
Ticket de soporte []

OmarYepez29 committed:
[MIG]Refactorización de funciones de calculo de impuestos.

Link:https://binaural.odoo.com/web#id=55261&cids=2&menu_id=257&action=341&model=project.task&view_type=form

OmarYepez29 committed:
[DEL]Eliminado _loggerr.warning

OmarYepez29 committed:
[MIG]Se descomenta los campos que eran dependientes de tax totals.

Se descomentan los campos que eran dependientes de tax totals,se cambia
a que campo de tax_totals hacen referencia.
Link:https://binaural.odoo.com/web#id=55235&cids=2&menu_id=257&action=341&model=project.task&view_type=form

Ivan-Binaural committed:
MIG #55255: l10n_ve_sale

Problema: Migracion de este modulo a la version 18

Solución:

Tarea (Link):

Tarea de proyecto [x]
Ticket de soporte []

bin-daldana commented on l10n_ve_accountant/__manifest__.py:
```diff
@@ -31,7 +31,7 @@
         "views/res_partner.xml",
         "views/res_currency_views.xml",
-        "views/ir_property.xml",
+        # "views/ir_property.xml",
```
Se queda comentado, por que?

OmarYepez29 replied:
@bin-daldana El ir.property a pesar de que sigue existiendo la vista,odoo no la encuentra

bin-daldana commented on l10n_ve_accountant/models/account_move.py:
```diff
@@ -4,7 +4,8 @@
 from odoo.exceptions import UserError, ValidationError
-from odoo.tools import drop_index, float_compare, index_exists
+from odoo.tools import float_compare, index_exists
+from odoo.tools.sql import drop_index
```
Considerable en caso de ser cierto

OmarYepez29 replied:
<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/4e74b20a-3eea-44d8-978a-f6c48a841a71" />
Es usado aqui

bin-daldana commented on l10n_ve_accountant/models/account_move.py:
```diff
@@ -900,59 +904,59 @@ def action_post(self):
-                            **invoice.needed_terms[key],
-                            "foreign_balance": sign * invoice.foreign_total_billed,
-                        }
+        # for invoice in self:
```
Que paso por aqui?

bin-daldana commented on l10n_ve_accountant/models/account_move_line.py:
```diff
@@ -442,48 +442,49 @@ def abs_amount_lines_ids_adjust(self):
-                    if tax["tax_repartition_line_id"] == key["tax_repartition_line_id"]:
-                        line.compute_all_tax[key]["foreign_balance"] = tax["amount"]
-        return res
+    #TODO:deprecated
```
cambio por otra funcion, compute, algo?

OmarYepez29 replied:
Odoo se la peino wn
<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/694e1024-3023-47b5-a6fb-0eec03fb1670" />

bin-daldana commented on l10n_ve_accountant/views/account_move.xml:
```diff
@@ -5,10 +5,10 @@
         <field name="arch" type="xml">
             <xpath expr="//header" position="after">
-                <field name="foreign_balance" invisible="1" />
+                <!-- <field name="foreign_balance" invisible="1" />
```
?????

bin-daldana commented on l10n_ve_contact/__manifest__.py:
```diff
@@ -8,7 +8,7 @@
     "category": "Contacts/Contacts",
     "version": "18.0.0.0.3",
-    "depends": ["base", "contacts", "l10n_ve_rate", "l10n_ve_location"],
+    "depends": ["base", "contacts", "account", "l10n_ve_rate", "l10n_ve_location"],
```
Porque se añade la dependencia?

bin-daldana reviewed (changes requested):
RC

Ivan-Binaural committed:
Merge branch '18-mig-l10n_ve_accountant-#55235' into l10n_ve_sale_MIG#55255

Ivan-Binaural committed:
Merge branch 'l10n_ve_stock_MIG_55257_migration_stock_module' into l10n_ve_sale_MIG#55255

Ivan-Binaural committed:
MIG [#55255](https://github.com/binaural-dev/odoo-venezuela/issues/55255): l10n_ve_sale

Problema: Migracion de este modulo a la version 18

Solución:

Tarea (Link):https://binaural.odoo.com/web/#id=55255&cids=2&menu_id=975&action=341&model=project.task&view_type=form

Tarea de proyecto [x]
Ticket de soporte []

OmarYepez29 committed:
WIP Descomentando funcion

OmarYepez29 committed:
Merge remote-tracking branch 'origin/l10n_ve_sale_MIG#55255' into 18-mig-l10n_ve_accountant-#55235

OmarYepez29 committed:
[WIP]Correciones en flujos de tax

OmarYepez29 committed:
[WIP]Adaptando funcionalidad de tax_totals a purchase.order y sale.order

OmarYepez29 committed:
[WIP]Añadida validacion para evitar busqueda de record  si no viene active_model

OmarYepez29 committed:
Merge branch 'dev-18' into 18-mig-l10n_ve_accountant-#55235

OmarYepez29 committed:
[ADD]Añadiendo unit test_accountant
Se añaden los unitest de alguna de las funcionalidades de test_accountant
Link:https://binaural.odoo.com/web#id=55235&cids=2&menu_id=257&action=341&model=project.task&view_type=form

OmarYepez29 committed:
Merge branch '18-mig-l10n_ve_accountant-#55235' of github.com:binaural-dev/odoo-venezuela into 18-mig-l10n_ve_accountant-#55235

OmarYepez29 committed:
[IMP]Correcion al compute de foreign_total_billed

OmarYepez29 committed:
[MIG]Se corrije el componente segun los cambios de odoo 18.
Link:https://binaural.odoo.com/web#id=55261&cids=2&menu_id=257&action=341&model=project.task&view_type=form

OmarYepez29 committed:
[IMP]Se refactoriza función que odoo cambio.

Se cambia el _get_total_amount_wizard_currency_to_full_reconcile a
_get_total_amounts_to_pay debido a cambios en la funcionalidad base de
odoo.

OmarYepez29 committed:
[FIX]Se corrije el mispelling de currency_foreign_id a foreign_currency_id.

OmarYepez29 committed:
[REF]Refactor de l10n_ve_tax.

Se mueve todo el comportamiento que antes existia en
l10n_ve_tax(Venezuela Impuestos) a l10n_ve_accountant(Venezuela
Contabilidad),debido a que existia una recuerrencia circular entre ambos
modulos.

OmarYepez29 committed:
Removes dependency on `l10n_ve_tax` module

Eliminates the `l10n_ve_tax` module from the dependency lists of multiple modules, including `l10n_ve_igtf`, `l10n_ve_invoice`, and `l10n_ve_sale`.

Replaces it with `l10n_ve_accountant` in `l10n_ve_tax_payer`, likely to streamline dependencies and improve modularity or compatibility.

OmarYepez29 committed:
[FIX]Correjido flujo de Libro de Compras y Ventas.

OmarYepez29 committed:
Refactors foreign currency handling and improves tax computations

Renames and updates methods to enhance clarity and align with naming conventions, improving the handling of foreign currency base and tax lines. Replaces redundant logging with cleaner logic to streamline code and reduce noise.

Introduces additional fields (`deferred_start_date`, `deferred_end_date`) for base line preparation, enhancing data granularity. Adjusts the handling of foreign balance calculations and tax lines to ensure proper synchronization and alignment with the main accounting logic.

Improves consistency in comments and variable naming for better readability and maintainability. Reduces code duplication by consolidating logic for preparing base lines and tax details.

OmarYepez29 committed:
[DEL]Borrada logica duplicada

OmarYepez29 committed:
[IMP]Se corrije update de apuntes contables de tipo impuesto

OmarYepez29 committed:
[FIX]Se corrije la forma de construir el libro de ventas tomando en consideracion el refactor de tax totals.

Se corrije el uso de los tax totals a la nueva version de odoo 18

OmarYepez29 committed:
[FIX]Fixes incorrect monetary formatting in tax totals
Replaces the use of a hardcoded formatting property with a
function to ensure proper monetary formatting of tax amounts
in the specified currency.
